tests/bsim/bluetooth/mesh/src/test_provision.c

```diff
@@ -1739,35 +1769,35 @@ static void test_device_pb_remote_server_ncrp_second_time(void)
 }
 #endif /* IS_RPR_PRESENT */
 
-#define TEST_CASE(role, name, description)                                     \
-	{                                                                      \
-		.test_id = "prov_" #role "_" #name, .test_descr = description, \
-		.test_post_init_f = test_##role##_init,                        \
-		.test_tick_f = bt_mesh_test_timeout,                           \
-		.test_main_f = test_##role##_##name,                           \
-		.test_delete_f = test_terminate                                \
-	}
-#define TEST_CASE_WBACKCHANNEL(role, name, description)                                     \
-	{                                                                      \
-		.test_id = "prov_" #role "_" #name, .test_descr = description, \
-		.test_post_init_f = test_##role##_init,                        \
-		.test_pre_init_f = test_back_channel_pre_init,                 \
-		.test_tick_f = bt_mesh_test_timeout,                           \
-		.test_main_f = test_##role##_##name,                           \
-		.test_delete_f = test_terminate                                \
-	}
+/* Test cases by default will run over PB_ADV*/
+#define TEST_CASE(role, name, description)                                                         \
+	{.test_id = "prov_" #role "_" #name,                                                       \
+	 .test_descr = description,                                                                \
+	 .test_args_f = test_args_parse,							   \
+	 .test_post_init_f = test_##role##_init,                                                   \
+	 .test_tick_f = bt_mesh_test_timeout,                                                      \
+	 .test_main_f = test_##role##_##name,                                                      \
+	 .test_delete_f = test_terminate}
+/* Test cases that will run over either PB_ADV or PB_GATT */
+#define TEST_CASE_WBACKCHANNEL(role, name, description)                                    \
+	{.test_id = "prov_" #role "_" #name,                                           \
+	 .test_descr = description,                                                                \
+	 .test_args_f = test_args_parse,							   \
+	 .test_post_init_f = test_##role##_init,                                                   \
+	 .test_pre_init_f = test_back_channel_pre_init,                                            \
+	 .test_tick_f = bt_mesh_test_timeout,                                                      \
+	 .test_main_f = test_##role##_##name,                                                      \
+	 .test_delete_f = test_terminate}
 
 static const struct bst_test_instance test_connect[] = {
-	TEST_CASE(device, pb_adv_no_oob,
```

Comment:
Really do not like such changes. It shuffles strings forth and back without any reasons. It actually hides the real change.
Please keep sequencing the same and change the only names to see what has been actually changed.

Reply:
I can't say I fully understand the comment; what the issue. However, I think adding pb-gatt and using the same test cases for both bearer necessitates removing pb-adv from related cases.